lib/simulation/openfisca/mapping/index.js

```diff
@@ -40,52 +40,63 @@ var applyRessources = function(src, dest, schema, dateDeValeur) {
 
     var ressourcesByType = _.groupBy(src.ressources, 'type');
     _.forEach(schema, function(definition, destKey) {
-        var srcKey, fn;
+        var srcKeys, fn;
         if (_.isString(definition)) {
-            srcKey = definition;
+            srcKeys = [definition];
+        } else if(_.isArray(definition)) {
+            srcKeys = definition;
         } else {
-            srcKey = definition.src;
+            srcKeys = [definition.src];
             fn = definition.fn;
         }
-
-        if (ressourcesByType[srcKey]) {
-            var result = {};
-            var agr = {};
-
-            _.forEach(ressourcesByType[srcKey], function(item) {
-                if (item.periode) {
-                    // Période d'origine
-                    if (!result[item.periode]) result[item.periode] = 0;
-                    result[item.periode] += item.montant;
-
-                    // Aggrégats
-                    _.keys(agregats).forEach(function (agregatKey) {
-                        if (periodeCompriseDans(agregatKey, moment(item.periode, 'YYYY-MM'))) {
-                            if (!(agregatKey in agr)) agr[agregatKey] = 0;
-                            agr[agregatKey] += fn ? fn(item.montant) : item.montant;
-                        }
-                    });
+        _.forEach(srcKeys, function(srcKey){
+
+            if (ressourcesByType[srcKey]) {
+                var result = {};
+                var agr = {};
+
+                _.forEach(ressourcesByType[srcKey], function(item) {
```

Comment:
This is getting pretty deep in nesting. Better use a helper function (see lines 142-149 for an example).